diff -up libkexiv2-18.12.0/cmake/modules/FindExiv2.cmake.omv~ libkexiv2-18.12.0/cmake/modules/FindExiv2.cmake
--- libkexiv2-18.12.0/cmake/modules/FindExiv2.cmake.omv~	2018-12-24 01:35:39.351857788 +0100
+++ libkexiv2-18.12.0/cmake/modules/FindExiv2.cmake	2018-12-24 01:36:00.105155517 +0100
@@ -57,7 +57,7 @@ find_library(EXIV2_LIBRARY NAMES exiv2 l
 
 # Get the version number from exiv2/version.hpp and store it in the cache:
 if(EXIV2_INCLUDE_DIR  AND NOT  EXIV2_VERSION)
-  file(READ ${EXIV2_INCLUDE_DIR}/exiv2/version.hpp EXIV2_VERSION_CONTENT)
+  file(READ ${EXIV2_INCLUDE_DIR}/exiv2/exv_conf.h EXIV2_VERSION_CONTENT)
   string(REGEX MATCH "#define EXIV2_MAJOR_VERSION +\\( *([0-9]+) *\\)"  _dummy "${EXIV2_VERSION_CONTENT}")
   set(EXIV2_VERSION_MAJOR "${CMAKE_MATCH_1}")
 
diff -up libkexiv2-18.12.0/src/kexiv2.cpp.omv~ libkexiv2-18.12.0/src/kexiv2.cpp
--- libkexiv2-18.12.0/src/kexiv2.cpp.omv~	2018-12-24 01:39:21.959046645 +0100
+++ libkexiv2-18.12.0/src/kexiv2.cpp	2018-12-24 01:40:58.297424055 +0100
@@ -27,6 +27,7 @@
 
 #include "kexiv2.h"
 #include "kexiv2_p.h"
+#include <exiv2/version.hpp>
 
 // Local includes
 
diff -up libkexiv2-18.12.0/src/kexiv2exif.cpp.omv~ libkexiv2-18.12.0/src/kexiv2exif.cpp
--- libkexiv2-18.12.0/src/kexiv2exif.cpp.omv~	2018-12-24 01:41:17.961705049 +0100
+++ libkexiv2-18.12.0/src/kexiv2exif.cpp	2018-12-24 01:43:08.285280677 +0100
@@ -987,7 +987,7 @@ bool KExiv2::setTiffThumbnail(const QIma
 
         if (pos == d->exifMetadata().end() || pos->count() != 1 || pos->toLong() != 0)
         {
-            throw Exiv2::Error(1, "Exif.Image.NewSubfileType missing or not set as main image");
+            throw Exiv2::Error(Exiv2::kerErrorMessage, "Exif.Image.NewSubfileType missing or not set as main image");
         }
 
         // Remove sub-IFD tags
diff -up libkexiv2-18.12.0/src/kexiv2_p.cpp.omv~ libkexiv2-18.12.0/src/kexiv2_p.cpp
--- libkexiv2-18.12.0/src/kexiv2_p.cpp.omv~	2018-12-24 01:40:02.499626441 +0100
+++ libkexiv2-18.12.0/src/kexiv2_p.cpp	2018-12-24 01:40:27.712986911 +0100
@@ -40,6 +40,8 @@ extern "C"
 #endif
 }
 
+#include <exiv2/version.hpp>
+
 // Qt includes
 
 #include <QTextCodec>
